test/integration_test.rb

```diff
@@ -229,6 +235,46 @@ def test_mixed_borders
     end
     @workbook.apply_styles
     assert_equal 8, @workbook.styled_cells.count
-    assert_equal 6, @workbook.style_index.count
+    assert_equal 6, @workbook.styles.style_index.count
+  end
+
+  def test_merge_styles_1
+    @filename = 'merge_styles_1'
+    @workbook.add_worksheet do |sheet|
+      sheet.add_row
+      sheet.add_row ['', '1', '2', '3']
+      sheet.add_row ['', '4', '5', '6']
+      sheet.add_row ['', '7', '8', '9']
+      sheet.add_style 'B2:D4', b: true
+      sheet.add_style 'B2:D4', border: { style: :thin, color: '000000' }
+    end
+    @workbook.apply_styles
```

Comment:
I see that both bold font face and border are being applied which is good. Unfortunately, the border is being drawn around every cell. In the original implementation the border is only added to the interior of the selected range (as in the example on the readme).

![border_issue](https://cloud.githubusercontent.com/assets/4750367/18030362/f28871de-6c80-11e6-9ef7-156a40dc07c8.png)